win32ss/user/ntuser/cursoricon.c

```diff
@@ -264,13 +264,15 @@ BOOL UserSetCursorPos( INT x, INT y, DWORD flags, ULONG_PTR dwExtraInfo, BOOL Ho
     pt.x = x;
     pt.y = y;
 
-    /* 1. Generate a mouse move message, this sets the htEx and Track Window too. */
-    Msg.message = WM_MOUSEMOVE;
-    Msg.wParam = UserGetMouseButtonsState();
-    Msg.lParam = MAKELPARAM(x, y);
-    Msg.pt = pt;
-    co_MsqInsertMouseMessage(&Msg, flags, dwExtraInfo, Hook);
-
+    if ((gpsi->ptCursor.x != x) || (gpsi->ptCursor.y != y))
```

Comment:
What about an early return?
```suggestion
    if (pt == gpsi->ptCursor)
        return TRUE;
```

Reply:
not sure this compiles (equality test between _structures_)

Reply:
Right, modified.

Reply:
This would not take care of setting the last global pointer to server info (gpsi) statement here:
```
    gpsi->ptCursor = pt;
```
Though this statement could be moved above the 'if'. I prefer that there be fewer 'return's.

Reply:
> This would not take care of setting the last global pointer to server info (gpsi) statement

Why is that needed when the values are identical?

> I prefer that there be fewer 'return's.

ROS coding style prefers `return`s.

Reply:
Please, could you answer my question?

Reply:
I have reviewed the "Programming Guidelines" here https://reactos.org/wiki/Programming_Guidelines.
I saw no reference to "ROS coding style prefers ```return ```s.
My preference is for a single return point at the end of the function. It is just my preference.

Reply:
The question is "Why is that needed when the values are identical?".

> no reference to "ROS coding style prefers `return `s."

Read https://reactos.org/wiki/Coding_Style#Control_structures

Reply:
@SergeGautherie, Thank you for pointing me to the reference that you mentioned previously. I see that the reason for this recommendation is to reduce the indentation levels and I agree that if this code were modified as you suggest, we could remove the open brace (Line 268) and close brace (Line 275) with having only added one line with the ```return TRUE;```. I see your point here. Thanks again.
Let's leave it up to @HBelusca or another developer based on whoever finally decides to commit this as to what direction they would like to have this code take.

Reply:
You have not answered the question about old line 275 yet.

Reply:
Submitted as  #6770.